spring-cloud-gateway-core/src/main/java/org/springframework/cloud/gateway/config/GatewayAutoConfiguration.java

```diff
@@ -585,12 +588,14 @@ else if (ssl.isUseInsecureTrustManager()) {
 								.trustManager(InsecureTrustManagerFactory.INSTANCE);
 					}
 
-					if (ssl.getKeyCertChain() != null
-							&& ssl.getKeyCertChain().length() > 0 && ssl.getKey() != null
-							&& ssl.getKey().length() > 0) {
-						sslContextBuilder = sslContextBuilder.keyManager(
-								new File(ssl.getKeyCertChain()), new File(ssl.getKey()),
-								ssl.getKeyPassword());
+					if (ssl.getKeyStore() != null && ssl.getKeyStore().length() > 0) {
+						try {
+							sslContextBuilder = sslContextBuilder
+									.keyManager(ssl.getKeyManagerFactory());
```

Comment:
Looks like we could maybe replace the `if()` statement above and encapsulate it in the `ssl.getKeyManagerFactory()` method?

Reply:
DId Suggested Changes